src/rust/src/x509/certificate.rs

```diff
@@ -319,6 +319,98 @@ impl Certificate {
             },
         )
     }
+
+    fn verify_signed_by<'p>(
```

Comment:
Should we be checking that issuer.subject matches our subject?

Reply:
I asked this question in a comment on the PR too 😄 I'm fine with doing that, although then maybe this should be called `verify_issued_by`?